drivers/dose/dose.c

```diff
@@ -191,9 +191,9 @@ static dose_signal_t state_transit_send(dose_t *ctx, dose_signal_t signal)
     /* Don't trace any END octets ... the timeout or the END signal
      * will bring us back to the BLOCKED state after _send has emitted
      * its last octet. */
-
+#ifndef MODULE_PERIPH_UART_COLLISION
```

Comment:
If I am not mistaken, not setting the timeout here should be accompanied by explicitly leaving the "SEND" state after a collision in `_send(...)` by calling `state(ctx, DOSE_SIGNAL_END);`.